src/config/mod.rs

```diff
@@ -1992,7 +1992,9 @@ impl RaftEngineConfig {
             // As the async-io is enabled by default (raftstore.store_io_pool_size == 1),
             // testing records shows that using 4kb as the default value can achieve
             // better performance and reduce the IO overhead.
-            self.mut_config().batch_compression_threshold = RaftEngineReadableSize(std::cmp::min(
+            // Meanwhile, the batch_compression_threshold cannot be modified dynamically if
+            // the threads count of async-io are changed manually.
+            self.mut_config().batch_compression_threshold = RaftEngineReadableSize(std::cmp::max(
```

Comment:
So does this means explicitly set config will also be changed? E.g, if user set the `batch-compression-threshold` to 16kb, it will be optimized to 8kb, is this expected behavior?